src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -194,7 +188,7 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
     {
         // Update NameMap with our parents.
         // Starting with this means we're doing a top-down approach.
-        FAIL_FAST_IF(_parents.size() > 1);
+        assert(_parents.size() <= 1);

```

Comment:
I personally believe `assert` should be preferred over `FAIL_FAST_IF` for "soft-assertions".
What I mean is basically assertions, that if they break don't lead to dangerously incorrect behavior. This is the case here: If `_parents` did have more than 1 member, it would IMO be preferable to rather do the slightly wrong thing, as opposed to crashing the app and causing the user to loose their data.

Reply:
I think @DHowett had some thoughts on removing `FAIL_FAST_IF`s from the codebase. Particularly regarding all of the ones in `Viewport` that caused a11y to shut down the whole app all the time (my fault). I don't recall what the resolution on that conversation was though.

That said, I'm ok with replacing `FAIL_FAST_IF`s with `assert`. It's compiled out of release builds, and if the condition isn't met in debug builds, it's treated like a `DebugBreak()` right?

Reply:
Yeah `assert()` calls `_wassert()`, which creates the famous 3-button-dialog where you can choose what you want to do (debug-break, abort, or continue IIRC).

Reply:
I don't know everywhere `IInheritable` is used, but it looks like at in `GlobalAppSettings`, `TerminalSettings`, and here in `ActionMap` it seems to enforced that there is at most one parent. If the ability to have multiple parents isn't actually used (I see some things with `Profile` but it is hard to tell if multiple direct parents is actually utilized compared to a chain of parents), wouldn't the better overall solution be to change IInheritable to have `std::optional<com_ptr<T>> _parent` instead?

Reply:
@Rosefield Unfortunately profiles _can_ have multiple parents, which is why this is so weird and tricky 😕